server/src/auth/index.ts

```diff
@@ -10,6 +13,30 @@ import { getDb } from '../database';
 import { configureAzureAuth } from './azure';
 import { configureGoogleOAuth } from './google-oauth';
 
+function basicAuth(req: Request): { name: string; pass: string } {
+  const string = req?.headers?.authorization;
+  if (!string) return;
+
+  const match = /^ *(?:[Bb][Aa][Ss][Ii][Cc]) +([A-Za-z0-9._~+/-]+=*) *$/.exec(
+    string,
+  );
+
+  if (!match) return;
+
+  // decode user pass
+  const userPass = /^([^:]*):(.*)$/.exec(
```

Comment:
[nit] This works fine, but wouldn't also just splitting by colon work here? :thinking: You could also easily destructure username and password from it in the same go:
```ts
const [user, pass] = Buffer.from(match[1], 'base64').toString().split(':');
```